src/org/pentaho/reporting/platform/plugin/async/PersistReportToJcrTask.java

```diff
@@ -89,14 +91,25 @@
 
       int copy = 1;
 
-      while ( dataLocation.exists( targetName ) ) {
-        targetName = String.format( FORMAT, cleanFileName, copy, targetExt );
-        copy++;
+      final OutputStream outputStream;
+
+      synchronized ( FORMAT ) {
```

Comment:
Instead of using a static lock (JVM wide lock-downs are sooo 20th century!), can you lock on the Repository-File instead?

You can use org.pentaho.platform.repository2.unified.jcr.ILockHelper's lockFile and unlockFile method for that. You get an instance of that class via the PentahoSystem#get method, as usual. Use "outputFolder.getId()" for as argument for the locking.

According to the documentation on DefaultLockhelper, this will create a deep lock (covers the node and all descendent nodes). 

Be aware that the lock will survive the session, so a try-finally is mandatory.